app/components/syntax-highlighted-diff.ts

```diff
@@ -161,27 +184,62 @@ export default class SyntaxHighlightedDiffComponent extends Component<Signature>
   }
 
   highlightCode = task({ keepLatest: true }, async (): Promise<void> => {
-    if (this.isDarkMode === undefined) {
-      return; // Not ready to highlight yet
+    // Read all dependent parameters into local constants
+    const {
+      codeWithoutDiffMarkers,
+      args: { code, language, forceDarkTheme },
+      darkMode: { isEnabled: isDarkMode },
+    } = this;
+
+    // Return if nothing changed since last render
+    if (
+      this.#lastHighlightCodeParams.compareWithParams({
+        isDarkMode,
+        forceDarkTheme,
+        language,
+        codeWithoutDiffMarkers,
+      })
+    ) {
+      return;
```

Comment:
What if `@code` has changed, but nothing else has? In that case don't we still need to run highlighting?

Reply:
In this case `codeWithoutDiffMarkers` will change and trigger the update

Reply:
Ah, didn't notice - all good!